Lib/glyphsLib/__main__.py

```diff
@@ -14,48 +14,185 @@
+        help=("The path to write the Glyphs file to."),
+    )
+
+    group = parser.add_argument_group('Roundtripping between UFOs and Glyphs')
+    group.add_argument(
+        "--no-preserve-glyphsapp-metadata",
+        action="store_false",
+        help=("Skip preserving Glyphs metadata in master UFOs and designspace "
+              "file, which would be used to minimize differences when "
+              "roundtripping between Glyphs and UFOs."),
+    )
+    group.add_argument(
+        "--enable-last-change",
+        action="store_false",
+        help=("Store modification timestamp in glyphs. Unnecessary "
+              "when using Git."),
+    )
+    group.add_argument(
+        "--enable-automatic-alignment",
+        action="store_false",
+        help=("Enable automatic alignment of components in glyphs."),
+    )
+    options = parser.parse_args(args)
+
+    sources = options.designspace_file_or_UFOs
+    if len(sources) == 1 and sources[0].endswith(
+            ".designspace") and os.path.isfile(sources[0]):
+        designspace_file = sources[0]
+        designspace = fontTools.designspaceLib.DesignSpaceDocument()
+        designspace.read(designspace_file)
+        object_to_read = designspace
+    elif all(
+            source.endswith(".ufo") and os.path.isdir(source)
+            for source in sources):
+        ufos = [defcon.Font(source) for source in sources]
+        ufos.sort(
+            key=lambda ufo: [  # Order the masters by weight and width
+                ufo.info.openTypeOS2WeightClass, ufo.info.openTypeOS2WidthClass],
+        )
+        object_to_read = ufos
+    else:
+        print(
+            "Please specify just one designspace file *or* one or more "
+            "UFOs. They must end in '.designspace' or '.ufo', respectively.",
+            file=sys.stderr)
+        return 1
+
+    font = glyphsLib.to_glyphs(
+        object_to_read,
+        minimize_ufo_diffs=options.no_preserve_glyphsapp_metadata)
+
+    # Make the Glyphs file more suitable for roundtrip:
+    font.customParameters["Disable Last Change"] = options.enable_last_change
+    font.disablesAutomaticAlignment = options.enable_automatic_alignment
+
+    if options.output_path:
+        font.save(options.output_path)
+    else:
+        if locals().get("designspace_file"):
+            filename_to_write = os.path.splitext(
+                designspace_file)[0] + ".glyphs"
         else:
-            glyphsLib.build_instances(opt.glyphs, opt.masters, opt.instances,
-                                      round_geometry=opt.round_instances)
+            filename_to_write = os.path.join(
+                os.path.dirname(sources[0]),
+                font.familyName.replace(" ", "") + ".glyphs")
+        font.save(filename_to_write)
+
+    return 0
+
 
-if __name__ == '__main__':
-    main(sys.argv[1:])
+if __name__ == "__main__":
+    print("Invoke either through glyphs2ufo or ufo2glyphs.", file=sys.stderr)
+    sys.exit(1)
```

Comment:
the purpose of `__main__` is to be able to run a package (i.e. folder with `__init__.py`) as a module, like `python -m glyphsLib`. I would like to keep this feature.
I suggest you move the two entry points to an importable module, `glyphsLib.cli`, and in `glyphsLib.__main__` you reintroduce a `main` function that calls either `glyphs2ufo` or `ufo2glyphs` based on the first argument, like a sub-command, so one can still do `glyphs -m glyphsLib [glyphs2ufo|ufo2glyphs] ...`